### PR TITLE
Create CITATION.cff

### PR DESCRIPTION
**Pull request by jhamman**
This adds a new file to the root of the Xarray repository, `CITATION.cff`. GitHub recently added [support](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files) for citation files and adding this file will add a UI feature to the Xarray GitHub repo.

The author list is based on the latest Zenodo release (0.20.1) and I did  my best to find everyone's ORCIDs.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+- family-names: "Hatfield-Dodds"
+  given-names: "Zac"
+  orcid: "https://orcid.org/0000-0002-8646-8362"
+- family-names: "Abernathey"
+  given-names: "Ryan"
+  orcid: "https://orcid.org/0000-0001-5999-4917"
+- family-names: "Bovy"
+  given-names: "Benoît"
+- family-names: "Motani"
+  given-names: "John"
```
Comment (TomNicholas) [inline CITATION.cff:70]:
```suggestion
- family-names: "Omotani"
  given-names: "John"
  orcid: "https://orcid.org/0000-0002-3156-8227"
```

Thanks @jhamman !

(@johnomotani FYI)

**Comment by jhamman** [comment]
I plan to merge this tomorrow unless I hear otherwise.

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  given-names: "Thomas"
+  orcid: "https://orcid.org/0000-0002-2176-0530"
+- family-names: "Kluyver"
+  given-names: "Thomas"
+  orcid: "https://orcid.org/0000-0003-4020-6364"
+- family-names: "Westling"
+  given-names: "Jimmy"
+- family-names: "Monroe"
+  given-names: "James"
+  orcid: "https://orcid.org/0000-0001-9098-6309"
```
Comment (mathause) [inline CITATION.cff:46]:
```suggestion
```
duplicate

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+- family-names: "Monroe"
```
Comment (mathause) [inline CITATION.cff:47]:
```suggestion
- family-names: "Munroe"
```
@jmunroe can you confirm?

**Review by mathause — COMMENTED** [review]
Thanks!

File: CITATION.cff (hunk 3 above)
Comment (jmunroe) [reply CITATION.cff]: Yes, that's now correct. Thanks!

Comment (jhamman) [reply CITATION.cff]: Sorry about that James!

**Pull request merged**